src/lcm/entry_point.py

```diff
@@ -59,196 +63,116 @@ def get_lcm_function(
-            continuous_choice_variables=list(_choice_grids),
+            continuous_choice_variables=tuple(state_choice_space.continuous_choices),
         )
-        compute_ccv_functions.append(compute_ccv)
 
-        compute_ccv_argmax = create_compute_conditional_continuation_policy(
+        compute_ccp = get_compute_conditional_continuation_policy(
             utility_and_feasibility=u_and_f,
-            continuous_choice_variables=list(_choice_grids),
+            continuous_choice_variables=tuple(state_choice_space.continuous_choices),
         )
-        compute_ccv_policy_functions.append(compute_ccv_argmax)
 
-        # create list of emax_calculators
-        # ==============================================================================
-        calculator = get_solve_discrete_problem(
-            random_utility_shock_type=_mod.random_utility_shocks,
-            variable_info=_mod.variable_info,
+        solve_discrete_problem = get_solve_discrete_problem_value(
+            random_utility_shock_type=internal_model.random_utility_shocks,
+            variable_info=internal_model.variable_info,
             is_last_period=is_last_period,
         )
-        emax_calculators.append(calculator)
+
+        state_choice_spaces[period] = state_choice_space
+        state_space_infos[period] = state_space_info
+        compute_ccv_functions[period] = compute_ccv
+        compute_ccp_functions[period] = compute_ccp
+        solve_discrete_problem_functions[period] = solve_discrete_problem
 
     # ==================================================================================
     # select requested solver and partial arguments into it
     # ==================================================================================
     _solve_model = partial(
         solve,
         state_choice_spaces=state_choice_spaces,
-        continuous_choice_grids=continuous_choice_grids,
         compute_ccv_functions=compute_ccv_functions,
-        emax_calculators=emax_calculators,
+        emax_calculators=solve_discrete_problem_functions,
         logger=logger,
     )
 
+    _next_state_simulate = get_next_state_function(
+        model=internal_model, target=Target.SIMULATE
+    )
+
     solve_model = jax.jit(_solve_model) if jit else _solve_model
+    next_state_simulate = jax.jit(_next_state_simulate) if jit else _next_state_simulate
 
-    _next_state_simulate = get_next_state_function(model=_mod, target=Target.SIMULATE)
     simulate_model = partial(
         simulate,
-        continuous_choice_grids=continuous_choice_grids,
-        compute_ccv_policy_functions=compute_ccv_policy_functions,
-        model=_mod,
-        next_state=jax.jit(_next_state_simulate),
+        compute_ccv_policy_functions=compute_ccp_functions,
+        model=internal_model,
+        next_state=next_state_simulate,  # type: ignore[arg-type]
+        logger=logger,
+    )
+
+    solve_and_simulate_model = partial(
+        solve_and_simulate,
```

Comment:
I would define the function `solve_and_simulate` here, I think. Or in a separate module of the same name. But I would not look for it in the `simulation.simulate` module...

If there was a clever way of telling `simulate_model` that the partialled argument to `simulate(vf_arr_dict)` will be the output of `solve_model` (2nd order partialling, I guess), that would be even better...